app/app.go

```diff
@@ -288,7 +288,7 @@ func NewTerraApp(
 		market.NewAppModule(appCodec, app.MarketKeeper, app.AccountKeeper, app.BankKeeper, app.OracleKeeper),
 		oracle.NewAppModule(appCodec, app.OracleKeeper, app.AccountKeeper, app.BankKeeper),
 		treasury.NewAppModule(appCodec, app.TreasuryKeeper),
-		wasm.NewAppModule(appCodec, &app.WasmKeeper, app.AccountKeeper, app.BankKeeper, app.interfaceRegistry),
+		wasm.NewAppModule(appCodec, &app.WasmKeeper, app.StakingKeeper, app.AccountKeeper, app.BankKeeper),
```

Comment:
I always wondered why the InterfaceRegistry was not extended to act as a general purpose DI container, it would probably make the app wiring a lot cleaner since you could essentially just pass the InterfaceRegistry to all modules and let them resolve any types via interfaces, proto msg or not. So I recon removing it all together from the wasm module which needs to be able to map a set or arbitrary instructions to cosmwasm, is a step in the wrong direction, however granted that the InterfaceRegistry currently has a limit on only accepting proto msgs it would probably required more work then its worth  :)

Reply:
Can I resolve?